lib/statsample-glm/glm/formula.rb

```diff
@@ -29,9 +29,11 @@ def parse_formula(form = :token)
 
       private
 
-      def add_contant_term_if_required
-        # TODO: Add support for constants
-        @tokens.unshift Token.new('1')
+      def manage_contant_term
+        @tokens.unshift Token.new('1') unless
+          @tokens.include?(Token.new('1')) ||
+          @tokens.include?(Token.new('0'))
+        @tokens.delete Token.new('0')
```

Comment:
I'd say that having `Token()` method could make such kind of code cleaner:
```ruby
@tokens.unshift Token('1') unless @tokens.include?(Token('1')) || @tokens.include?(Token('0'))
@tokens.delete Token('0')
```

(This approach also allows some optimizations, like `Token('1')` always returning the same, constant, object, instead of creating tons of same objects on each and every check "if something is `Token.new('1')`")
WDYT?